cpp/compiler.cc

```diff
@@ -235,8 +235,16 @@ CompiledGrammar MultiThreadCompileGrammar(
   // 1. All character class or character class star (with last_utf8_bytes=0, 1, 2, 3)
   // 2. All byte strings (with element_in_string=0, 1, 2, ...)
 
-  ThreadPool thread_pool(max_threads);
-  std::mutex adaptive_token_mask_cache_mutex;
+  // TODO(Charlie): Figure out how to support ThreadPool and std::mutex in WebAssembly.
+  // Only declare ThreadPool and mutex if max_threads > 1, so when max_threads = 1, we do
```

Comment:
we don't have to do that as threading in webassembly is tricky and needs webworkers